crates/wit-bindgen/src/lib.rs

```diff
@@ -162,18 +162,57 @@ impl Wasmtime {
                 gen.generate_trappable_error_types(TypeOwner::Interface(*id));
                 let iface = &resolve.interfaces[*id];
 
+                let wildcard = iface.functions.values().find(|func| "*" == &func.name);
+
+                if let Some(wildcard) = wildcard {
+                    uwriteln!(gen.src, "pub struct WildcardMatch {{");
+                    uwriteln!(gen.src, "name: String,");
+                    uwriteln!(gen.src, "func: wasmtime::component::TypedFunc<(");
+                    for (_, ty) in wildcard.params.iter() {
+                        gen.print_ty(ty, TypeMode::AllBorrowed("'_"));
+                        gen.push_str(", ");
+                    }
```

Comment:
I would recommend adding some codegen tests for wildcards with "interesting" types as I think that a wildcard function which takes a `string` parameter might break this since it'll be printed as `&str` which I think will complain about not having the right lifetime. (this is the motivation for storing a `Func` rather than a `TypedFunc` in these structures)